pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/ValueParser.java

```diff
@@ -0,0 +1,263 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.rule.properties;
+
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.ARRAY_FLAG;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.CLASS_METHOD_DELIMITER;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.METHOD_ARG_DELIMITER;
+import static net.sourceforge.pmd.lang.rule.properties.modules.MethodPropertyModule.METHOD_GROUP_DELIMITERS;
+
+import java.lang.reflect.Array;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sourceforge.pmd.util.ClassUtil;
+import net.sourceforge.pmd.util.StringUtil;
+
+/**
+ * Parses a value from a string.
+ *
+ * @param <U> The type of the value to parse
+ */
+// FUTURE @FunctionalInterface
+public interface ValueParser<U> {
```

Comment:
You're right, I'll move them. 
Thinking about it, maybe a better solution would be not to use a functional interface but a utility class with static parsing methods. That way, we use eg `ValueParsers.parseMethod(valueString)` which is shorter and more expressive than `METHOD_PARSER.valueOf(valueString)`, and `ValueParsers.parseMultiPropertyValues(valueString, DELIMITER, ValueParsers::parseMethod)`. We should end up with a bit less boilerplate, and a cleaner syntax. Plus I think it might be better style to define a bunch of static utility methods than a bunch of lambda constants :) We'd need find another name than `ValueParsers` though, maybe `ParseUtil`? I guess it should be package private too, not like our current interface